### PR TITLE
Create Climate Setting Schedule Form - name and schedule

### PR DESCRIPTION
**Pull request by Andrewjeska**
Screenshot 2023-09-12 at 5 44 21 PM

### CODE REVIEW

File: src/lib/ui/ClimateSettingForm/ClimateSettingScheduleForm.tsx [hunk 1]
```diff
@@ -38,7 +42,13 @@ export function ClimateSettingScheduleForm({
...
   onBack,
 }: Omit<ClimateSettingScheduleFormProps, 'className'>): JSX.Element {
-  const { control } = useForm()
+  const { control, watch } = useForm()
+
+  const deviceId = watch('deviceId', null)
+
+  const { device } = useDevice({
+    device_id: deviceId,
+  })
```
Comment (Andrewjeska) [inline src/lib/ui/ClimateSettingForm/ClimateSettingScheduleForm.tsx:51]: I'd like to let you override the react query params so I can do something like:
``` 
const { device } = useDevice({
    device_id: deviceId,
  },
	{enabled: Boolean(deviceId)}
)
```

Comment (razor-x) [reply src/lib/ui/ClimateSettingForm/ClimateSettingScheduleForm.tsx:51]: I think we can just split each step into its own component to handle this.

Comment (Andrewjeska) [reply src/lib/ui/ClimateSettingForm/ClimateSettingScheduleForm.tsx:51]: great idea!

Comment (Andrewjeska) [reply src/lib/ui/ClimateSettingForm/ClimateSettingScheduleForm.tsx]: So one hiccup, all the other steps do have the same heading, I did split it out though and it seems cleaner. 

I can put `<ContentHeader/>` above  `<Content/>`, that seems appropriate 😅

Comment (Andrewjeska) [reply src/lib/ui/ClimateSettingForm/ClimateSettingScheduleForm.tsx]: https://github.com/seamapi/react/pull/460/commits/68c485e177f171557e794df536a9edc09be21c3a

**Comment by Andrewjeska** [comment]
I'll take a crack at getting vscode to do the imports from the top level in a diff PR

File: src/lib/ui/ClimateSettingForm/ClimateSettingScheduleFormNameAndSchedule.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+import { Button } from '../Button.js'
+import { ContentHeader } from '../layout/ContentHeader.js'
+
+interface ClimateSettingScheduleFormNameAndScheduleProps {
+  control: Control<FieldValues, any>
+  deviceId: string
+  onBack: () => void
+  onCancel: (() => void) | undefined
+  onNext: () => void
+  className?: string
```
Comment (razor-x) [inline src/lib/ui/ClimateSettingForm/ClimateSettingScheduleFormNameAndSchedule.tsx:19]: Is this prop needed?

```suggestion
```

Comment (razor-x) [reply src/lib/ui/ClimateSettingForm/ClimateSettingScheduleFormNameAndSchedule.tsx:19]: or if it is required
```suggestion
  className: string | undefined
```

File: src/lib/ui/ClimateSettingForm/ClimateSettingScheduleFormDeviceSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+export function ClimateSettingScheduleFormDeviceSelect({
+  control,
+  onBack,
+  onSelect,
+}: ClimateSettingScheduleFormDeviceSelectProps): JSX.Element {
+  return (
+    <>
+      <ContentHeader title={t.addNewClimateSettingSchedule} onBack={onBack} />
```
Comment (razor-x) [inline src/lib/ui/ClimateSettingForm/ClimateSettingScheduleFormDeviceSelect.tsx:20]: title could be a prop passed in from the parent

File: src/lib/ui/ClimateSettingForm/ClimateSettingScheduleFormNameAndSchedule.tsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+                defaultValue=''
+                render={({ field }) => (
+                  <DateTimePicker {...field} size='large' />
+                )}
+              />
+            </FormField>
+          </div>
+        </div>
+        <div className='seam-actions'>
+          <Button onClick={onCancel ?? undefined}>{t.cancel}</Button>
```
Comment (razor-x) [inline src/lib/ui/ClimateSettingForm/ClimateSettingScheduleFormNameAndSchedule.tsx:93]: According to the type this looks unnecessary 
```suggestion
          <Button onClick={onCancel}>{t.cancel}</Button>
```

**Review by razor-x — APPROVED** [review]

File: src/lib/ui/ClimateSettingForm/ClimateSettingScheduleFormDeviceSelect.tsx (hunk 3 above)
Comment (Andrewjeska) [reply src/lib/ui/ClimateSettingForm/ClimateSettingScheduleFormDeviceSelect.tsx]: [e1c6eab](https://github.com/seamapi/react/pull/460/commits/e1c6eab7e4bc2547e8364137a8eee34e71a4408e)

File: src/lib/ui/ClimateSettingForm/ClimateSettingScheduleFormDeviceSelect.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import { type Control, Controller, type FieldValues } from 'react-hook-form'
+
+import { ContentHeader } from 'lib/ui/layout/ContentHeader.js'
+import { ThermostatSelect } from 'lib/ui/thermostat/ThermostatSelect.js'
+
+interface ClimateSettingScheduleFormDeviceSelectProps {
+  title: string
+  control: Control<FieldValues, any>
```
Comment (razor-x) [inline src/lib/ui/ClimateSettingForm/ClimateSettingScheduleFormDeviceSelect.tsx:8]: I think the second param should be optional unless we are using the context.
```suggestion
  control: Control<FieldValues>
```

File: src/lib/ui/ClimateSettingForm/ClimateSettingScheduleFormNameAndSchedule.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+import { Button } from 'lib/ui/Button.js'
+import { DateTimePicker } from 'lib/ui/DateTimePicker/DateTimePicker.js'
+import { FormField } from 'lib/ui/FormField.js'
+import { InputLabel } from 'lib/ui/InputLabel.js'
+import { ContentHeader } from 'lib/ui/layout/ContentHeader.js'
+import { TextField } from 'lib/ui/TextField/TextField.js'
+
+interface ClimateSettingScheduleFormNameAndScheduleProps {
+  title: string
+  control: Control<FieldValues, any>
```
Comment (razor-x) [inline src/lib/ui/ClimateSettingForm/ClimateSettingScheduleFormNameAndSchedule.tsx:13]:
```suggestion
  control: Control<FieldValues>
```

**Pull request merged**